### DISCUSSION TITLE
Super laggy with a large camera roll

### DISCUSSION THREAD

**Discussion by s-alad** [IacobIonut01/Gallery#377] [Q&A]
Is the app known to not support big camera rolls, or am I just unlucky?

**Comment by ako256** [comment]
It stutters even on my phone which has a snapdragon 8+ gen 1.
I don't think your phone is the problem.

**Reply by IacobIonut01** [reply]
Hi, how big of a camera roll are we talking about? I'm trying to replicate with 10k items on the Pixel 6 and no success

**Reply by ako256** [reply]
Hello
It is very random so I can't replicate it all the time as well, but sometimes, it stutters when there are a lot of videos whose thumbnails haven't been cached

Pressing on videos in general is also very glitchy 

https://github.com/IacobIonut01/Gallery/assets/118416042/cbd9cc01-9d8a-4e2a-8fce-28058c9d52be

Except these things, the app is really good, thanks!

**Reply by IacobIonut01** [reply]
Thanks, I guessed it's the video frame that makes it slower.

**Comment by IacobIonut01** [comment]
Try the latest nightly release

**Reply by ako256** [reply]
I feel like the random lags have become a lot infrequent but the video glitch is still present

**Reply by IacobIonut01** [reply]
It might be the best I can do until Jetpack Compose adds extra optimisation

**Reply by ako256** [reply]
This is truly sad, the app you've made is seriously on par with Google photos and apple photos but it's limited by this optimisation issue.

**Reply by ako256** [reply]
One thing I've noticed other apps do is first complete the expansion animation from the gallery to image view while loading the video behind the cached thumbnail image to kind of hide the jank. Is this possible?

**Reply by IacobIonut01** [reply]
> I feel like the random lags have become a lot infrequent but the video glitch is still present

Can you send a video when it happens?

**Reply by ako256** [reply]
It is the same as the previous video I sent, but maybe a bit faster. I've attached a new recording of it below

https://github.com/IacobIonut01/Gallery/assets/118416042/42c9d424-e604-4d73-a25c-d13a19df6e0a

**Reply by IacobIonut01** [reply]
I thought it was still while scrolling

**Reply by ako256** [reply]
It is but it's so rare now that I can't replicate it

**Reply by s-alad** [reply]
still getting having scroll lag even with nightly :(

**Comment by AkazaRenn** [comment]
https://github.com/user-attachments/assets/2457dfa3-78b5-4f28-afb9-b7b17dddaee4

I'm still feeling a bit of the lagging on 3.0.0. I think it results from a combination of low frame rate and bad animations curve, by comparing it with Simple Gallery (2nd half of the recording).

**Reply by IacobIonut01** [reply]
> https://github.com/user-attachments/assets/2457dfa3-78b5-4f28-afb9-b7b17dddaee4
> 
> I'm still feeling a bit of the lagging on 3.0.0. I think it results from a combination of low frame rate and bad animations curve, by comparing it with Simple Gallery (2nd half of the recording).

Try to disable the fancy blur in the settings, that might do it for you

**Reply by AkazaRenn** [reply]
I tried that but couldn't feel much difference...
